lib/spack/spack/analyzers/analyzerbase.py

```diff
@@ -0,0 +1,105 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+"""An analyzer base provides basic functions to run the analysis, save results,
+and (optionally) interact with a Spack Monitor
+"""
+
+import spack.monitor
+import spack.hooks
+import llnl.util.tty as tty
+import spack.util.path
+import spack.config
+
+import os
+
+
+def get_analyzer_dir(spec, analyzer_dir=None):
+    """Given a spec, return the directory to save analyzer results,
```

Comment:
We're not super religious about docstrings but FYI, if you want the docs to look nice, a one-line summary with a period followed by the rest of the text is what PEP257 says you're supposed to do: https://www.python.org/dev/peps/pep-0257/#multi-line-docstrings.   I think some doc formatters care.

Reply:
I'd like to do them right. I'll go through the PR (probably tomorrow, need to have dinner) and fix them up.

Reply:
Another question - what standard do you use for commits? I notice you do:

```
<category>: short message

longer message
```
And I'd like to do that too - I currently do something similar but without the category. I've seen you use "bugfix" as a category, but how do I know what the others are? (I'd guess there is docs too).